### ISSUE TITLE
Code quality: Improve strict 

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#1439]
## The problem

We have enabled [strict types](https://www.phptutorial.net/php-tutorial/php-strict_types/) already in https://github.com/PrivateBin/PrivateBin/pull/1350, however this lead to issues like https://github.com/PrivateBin/PrivateBin/issues/1435. It is an example of what can happen, if type casting goes wrong.

Note generally, if we have strict types, we should also declare what types we use, just as other programming languages. A discussion of advantages of strict typing vs loose (or no) typing (at all, what PHP did before) goes beyond the scope of this issue, but in short:
> Type hinting in method parameters or class properties is very useful and it helps you catch type errors before you run your code. 

https://dev.to/rocksheep/the-way-stricttypes-works-in-php-eb7

As [PHP nowadays supports type hints/type declarations](https://stackoverflow.com/a/29967975/5008962) as of PHP v7

## The solution

* [ ] first of all, check, if strict types are enabled in all files (excluding dependencies of course)
* [ ] step by step, convert the code [and add type declarations](https://www.php.net/manual/en/language.types.declarations.php) to the functions, both for arguments as well as
* [ ] test that everything still works as wanted, both via unit tests and manual tests
* [ ] AFAIK, you can then remove the PHPDoc parts that only explain the types (like return values or arguments). But let's test that they appear the same in [PHPDoc](https://phpdoc.org/).

Some hints:
* One can use the existing PHPdoc as a very good reference, at best it works the same and the documented type is really what the runtime really wants. However, there may be edge cases or cases where it is not "quite right" (as obviously it was never enforced by the [PHP] interpreter), so take it with a grain of salt!
* Especially for contributors: Start with one file! Or generally, take it step by step. You can easily do multiple PRs and start small by converting only one or two files to stricter types. It can be gradually done.

## Alternatives

Leave it as it is, it should stay compatible as PHP is very backwards-compatible. However, modern code and many other programming languages embrace type casting/strict/strong(er) typing, so this is IMHO a good way, we should follow.

## Additional context

* Bonus: Maybe we find a linter tool or can configure sth. to alert us about missing type declarations in the future? This would help:
  1. to discover missing declarations
  2. to enforce a new coding style that requires type declarations for PHP code

Also, as it is a feature starting with PHP 7, but which has been gradually enhanced, we obviously need to make sure it works with all supported PHP versions:
https://github.com/PrivateBin/PrivateBin/blob/d23bb748d45bb8cde8f3737b632d7ddd6a383e35/.github/workflows/tests.yml#L25

**Comment by elrido** [comment]
When and if we do this, we should test if we can also drop the docblock types at the same time (not the whole docblock, the description is still useful). docblocks frequently become outdated and no longer match what the function actually returns, while in strict mode we now can expect these type hints to be correct or fail during execution.

**Comment by rugk** [comment]
Yes I added this to the initial post already. Whether that can be done automatically, I'm not sure. (but yeah finding a tool would be cool)

**Comment by Ribas160** [comment]
Since the [minimum PHP version supported by this project is PHP 7.4](https://github.com/PrivateBin/PrivateBin/releases/tag/2.0.0), we can add type declarations not only to functions but also to [class properties](https://www.php.net/manual/en/language.oop5.properties.php).
